sancus-step/main.c

```diff
@@ -0,0 +1,76 @@
+#include <msp430.h>
+#include <stdio.h>
+#include <sancus/sm_support.h>
+#include <sancus_support/sm_io.h>
+#include <sancus_support/sancus_step.h>
+
+#define MAX_COUNTER (0xf)
+
+int counter = 0x0;
+int oldCounterValue = 0x0;
+
+/*
+ * Foo module
+ */
+
+DECLARE_SM(foo, 0x1234);
+
+void SM_ENTRY(foo) foo_enter(void)
+{
+    __asm__ __volatile__(
+        "mov #0x0, &counter\n\t"
+        "mov #0x1, &counter\n\t"
+        "mov #0x2, &counter\n\t"
+        "mov #0x3, &counter\n\t"
+        "mov #0x4, &counter\n\t"
+        "mov #0x5, &counter\n\t"
+        "mov #0x6, &counter\n\t"
+        "mov #0x7, &counter\n\t"
+        "mov #0x8, &counter\n\t"
+        "mov #0x9, &counter\n\t"
+        "mov #0xa, &counter\n\t"
+        "mov #0xb, &counter\n\t"
+        "mov #0xc, &counter\n\t"
+        "mov #0xd, &counter\n\t"
+        "mov #0xe, &counter\n\t"
+        "mov #0xf, &counter\n\t"
+        :::);
+}
+
+
+/*
+ * Untrusted context
+ */
+
+void checkCounter(void)
+{
+    if (counter != 0)
+    {
+        if (!(counter == oldCounterValue || counter == oldCounterValue + 1))
```

Comment:
nice! maybe it also makes sense to print a warning message here if zero-stepping is detected?